inc/class-core-sitemaps-provider.php

```diff
@@ -42,6 +42,23 @@ class Core_Sitemaps_Provider {
 	 */
 	public $slug = '';
 
+	/**
+	 * Set up relevant rewrite rules, actions, and filters.
+	 */
+	public function setup() {
+		// Set up rewrite rules and rendering callback.
+		add_rewrite_rule( $this->route, $this->rewrite_query(), 'top' );
+		add_action( 'template_redirect', array( $this, 'render_sitemap' ) );
+
+		// Set up async tasks related to calculating lastmod data.
+		add_action( 'core_sitemaps_calculate_lastmod', array( $this, 'calculate_sitemap_lastmod' ), 10, 3 );
+		add_action( 'core_sitemaps_update_lastmod_' . $this->slug, array( $this, 'update_lastmod_values' ) );
+
+		if ( ! wp_next_scheduled( 'core_sitemaps_update_lastmod_' . $this->slug ) && ! wp_installing() ) {
+			wp_schedule_event( time(), 'twicedaily', 'core_sitemaps_update_lastmod_' . $this->slug );
```

Comment:
Consider making the schedule `twicedaily` filterable, as depending on the use case spidering twice a day is not enough, eg news. (assuming the sitemap's last modified date influences whether the sitemap page should be explored by the crawler).